### ISSUE TITLE
pdf2image.exceptions.PDFInfoNotInstalledError: Unable to get page count. Is poppler installed and in PATH?

### ISSUE THREAD

**Issue by DankeF** [Belval/pdf2image#50]
I use a MAC, according to the README installed popple, PIP also installed pdf2image, but wrong in the code to run times: pdf2image. Exceptions. PDFInfoNotInstalledError: Unable to get page count. Is poppler installed and in the PATH?

![image](https://user-images.githubusercontent.com/29647627/52898159-81821a00-3215-11e9-92c1-482b97bec513.png)

**Comment by Belval** [comment]
Hi!

First, open a terminal and try to invoke pdfinfo manually to see if your installation was successful.

**Comment by DankeF** [comment]
@Belval 
Thank you for your reply. Pdfinfo can be output correctly。
![image](https://user-images.githubusercontent.com/29647627/52901397-ee110f00-323d-11e9-978b-0cd81fd34838.png)

**Comment by Belval** [comment]
Indeed, this is a probably a real bug then. Unfortunately the current code hides the actual exception.

If you would be so kind, please edit https://github.com/Belval/pdf2image/blob/master/pdf2image/pdf2image.py#L186 to print the actual exception so I'll be more able to help you. Something like:

```py
except Exception as ex:
    raise ex
```

Would do the trick.

You can edit it directly in `/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdf2image/pdf2image.py` (the path in your flask stacktrace)

**Comment by DankeF** [comment]
@Belval 
I edited pdf2image.py according to your instructions, and then ran the program to output this error: FileNotFoundError: [Errno 2] No such file or directory: 'pdfinfo': 'pdfinfo'
![image](https://user-images.githubusercontent.com/29647627/52908964-d4fa7380-32ba-11e9-8332-37869a1cb304.png)

**Comment by Belval** [comment]
I don't have a machine with OSX to test, but I think this could be related to flask. It is possible that it's running in a special instance, like a docker container? Try running your conversion code as a typical python script.

A good test would be to simply spawn a live python shell, import pdf2image, and convert the PDF in there.

**Comment by DankeF** [comment]
@Belval 
I tried to write a typical python script. It will still report an error. I am confused why I can't find the pdfinfo command.
![image](https://user-images.githubusercontent.com/29647627/52923573-7210e700-3363-11e9-9ba9-a5fa701b7545.png)
![image](https://user-images.githubusercontent.com/29647627/52923601-87861100-3363-11e9-8a6c-2eb05428c6ba.png)

**Comment by Belval** [comment]
That would probably boil down to how OSX deals with installation I suppose, maybe it needs to be installed in system PATH? Or there is a special process to make is accessible to other processes?

I'm really making blind guesses here as I have no way to test without a Mac.

**Comment by DankeF** [comment]
@Belval 
I found the cause of the problem:
PyCharm's terminal input pdfinfo command is not recognized.
![image](https://user-images.githubusercontent.com/29647627/53318269-abb89380-3909-11e9-902f-a6e23a74679e.png)

Typing pdfinfo into the MAC terminal is identifiable. 
![image](https://user-images.githubusercontent.com/29647627/53318311-d276ca00-3909-11e9-8bf7-e9fe0edc353b.png)

This means that the terminal in PyCharm cannot read the environment variable of ~/.bash_profile, I only need to run the program in the system terminal.

Thank you for every reply, let me get the idea.

**Comment by Belval** [comment]
Glad you got it sorted out!

I'll be closing this for now, as I do not think much can be done on `pdf2image` side to fix this.

**Comment by JaNi95** [comment]
[https://intellij-support.jetbrains.com/hc/en-us/community/posts/208567485-Pycharm-terminal-is-missing-part-of-PATH](url)

> The problem seems to happen only with "shell integration" enabled.
> 
> If I uncheck Tools > Terminal > "shell integration" terminal will load normally, with full $PATH and normal settings.

**Comment by gopikrishnareddy93** [comment]
We ran into this issue when we were building release package using jenkins, we were able to resolve this issue by updating the permissions on the poppler_binaries directory, execute the below  command to fix the permissions.

```sh
chmod ugo+rwx -R package/poppler_binaries
```

**Comment by NicoLivesey** [comment]
Hello,

I'm having the same problem in a docker container and I can't find the `package/poppler_binaries` directory in it. Do you have other insight to solve this problem ?

**Comment by glushakviktor** [comment]
I had same problems on Centos 7 server (on Ubuntu apt-get instead yum ).  
My solution:  

1. install pdfinfo (it is in package poppler-utils)
  
`sudo yum update -y`  
`sudo yum install -y poppler-utils`  

2. wrap try-except string 186 in pdf2image.py (my path is something  like that venv/lib/site-packages/pdf2image):   

``` 
try:    
    processes.append(
            (thread_output_file, Popen(args, env=env, stdout=PIPE, stderr=PIPE))
        )
except Exception as err:   
    print(err)
```

**Comment by TACHENZHICHAO** [comment]
> @Belval
> 我试图编写一个典型的python脚本。它仍然会报告错误。我很困惑为什么找不到pdfinfo命令。
> ![图片](https://user-images.githubusercontent.com/29647627/52923573-7210e700-3363-11e9-9ba9-a5fa701b7545.png)
> ![图片](https://user-images.githubusercontent.com/29647627/52923601-87861100-3363-11e9-8a6c-2eb05428c6ba.png)

Haven't you solved it yet?Download poppler-0.68.0_x86,and add path: ''C:\poppler-0.68.0\bin''.
Your Finished
